soroban-env-host/src/cost_runner/cost_types/host_mem_cmp.rs

```diff
@@ -0,0 +1,20 @@
+use soroban_env_common::Compare;
+
+use crate::{
+    budget::{AsBudget, CostType},
+    cost_runner::CostRunner,
+};
+
+pub struct HostMemCmpRun;
+impl CostRunner for HostMemCmpRun {
+    const COST_TYPE: CostType = CostType::HostMemCmp;
+    type SampleType = (Vec<u8>, Vec<u8>);
+
+    fn run_iter(host: &crate::Host, _iter: u64, sample: Self::SampleType) {
+        host.as_budget()
+            .compare(&sample.0.as_slice(), &sample.1.as_slice())
+            .unwrap();
+        // `forget` avoids deallocation of sample which artificially inflates the cost
+        std::mem::forget(sample)
```

Comment:
I'm a bit uneasy about this forget call. I mean, it works, but I am thinking another possible interface might be something like:

1. Change `run_iter` to return `Option<Self::Sample>` so that each cost type can choose whether it's most appropriate to _consume_ its sample or _return_ it.
2. Change the caller to pre-allocate an equal-sized Vec<Sample> to collect-up "non-consumed" samples from `run_iter`, and then free them all in a batch once the instrumentation machinery is turned off.

This would have the added benefit of making it less likely that the optimizer deletes a cost measurement altogether due to observing that "nobody uses the result of `run_iter`".

Reply:
I have fixed this by adding the `RecycledType` to every run type, please take another look.